gk/fib.c

```diff
@@ -1382,8 +1372,53 @@ check_prefix_locked(struct ip_prefix *prefix,
 	struct gk_fib *ip_prefix_fib;
 	struct gk_lpm *ltbl = &gk_conf->lpm_tbl;
 
-	if (action == GK_DROP || action == GK_FWD_GRANTOR)
+	if (action == GK_DROP || action == GK_FWD_GRANTOR) {
+		if (prefix->addr.proto == ETHER_TYPE_IPv4) {
+			uint8_t depth = prefix->len;
+			uint32_t next = 0;
+			int index = rte_lpm_rule_iterate(ltbl->lpm,
+				ntohl(prefix->addr.ip.v4.s_addr),
```

Comment:
Shouldn't we have a variable for the expression `ntohl(prefix->addr.ip.v4.s_addr)` since it's used here and at the end of each iteration of next loop?

If you agree, review `rte_lpm_is_rule_present()` later in the code.